### DISCUSSION TITLE
Doris 2.0.2版本无法添加rollup

### DISCUSSION THREAD

**Discussion by rotkang** [apache/doris#26028] [A - General / Q&A] [answered]
### 添加语法
`alter table cml.ads_rollup_agg1 add rollup rollup_p_dt(p_dt,rqst_cnt,estd_inc,rqst_id)`

### 错误原因
`SHOW ALTER TABLE MATERIALIZED VIEW FROM cml order by createtime desc;`
详细错误：
errCode = 2, detailMessage = rollup tasks failed on same tablet reach threshold [[ALTER], signature: 4182919, backendId: 10195, tablet id: 4182919, [ALTER], signature: 4182919, backendId: 10234, tablet id: 4182919], reason=task type: ALTER, status_code: INTERNAL_ERROR, status_message: [(172.22.64.34)[E-1602]SchemaChange check row nums failed], backendId: Backend [id=10234, host=172.22.64.34, heartbeatPort=9050, alive=true, lastStartTime=2023-10-10 18:39:31, tags: {location=default}], signature: 4182919

### 查询表DDL信息
drop table if exists cml.`ads_rollup_agg1`;
CREATE TABLE cml.`ads_rollup_agg1` (
  `p_dt` date NOT NULL COMMENT '统计日期',
  `p_hr` varchar(2) NULL COMMENT '小时',
  `billing_id` varchar NULL COMMENT '',
  `media_id` varchar NULL COMMENT '',
  `stgy_id` varchar NULL COMMENT '',
  `main_advs_id` varchar NULL COMMENT '',
  `ad_src_id` varchar NULL COMMENT '',
  `evt_nm` varchar(50) NULL COMMENT '',
  `evt_state` varchar(10) NULL COMMENT '',
  `rqst_cnt` bigint(20) SUM NULL DEFAULT "0" COMMENT '',
  `estd_inc` double SUM NULL DEFAULT "0" COMMENT '',
  `rqst_id` bitmap BITMAP_UNION NOT NULL COMMENT '请求ID'
) ENGINE=OLAP
AGGREGATE KEY(`p_dt`, `p_hr`, `billing_id`, `media_id`, `stgy_id`, `main_advs_id`, `ad_src_id`, `evt_nm`, `evt_state`)
COMMENT 'OLAP'
PARTITION BY RANGE(`p_dt`)()
DISTRIBUTED BY HASH(`p_dt`, `p_hr`) BUCKETS auto
;

### 表数据信息
一天表明细记录为：6.9亿
导入OLAP AGGREGATE模型后记录数为：35万（359533）条     rqst_id字段去重后UV为：约1亿（92573731）
查询时间过长
MySQL [ads]> select p_dt,bitmap_union_count(rqst_id) uv,sum(rqst_cnt) rqst_cnt
    -> from cml.ads_rollup_agg1
    -> where p_dt= '2023-10-25'
    -> group by p_dt;
+------------+----------+-----------+
| p_dt       | uv       | rqst_cnt  |
+------------+----------+-----------+
| 2023-10-25 | 92573731 | 694103199 |
+------------+----------+-----------+
1 row in set (2 min 29.50 sec)

请问是什么原因导致的？正常情况数据量少是能正常添加成功，数据量一大就添加失败，错误参见【错误原因】
先谢谢了。

**Comment by liugddx** [comment]
Have you done an upgrade?

**Reply by rotkang** [reply]
版本：Doris version doris-2.0.2-rc05-ae923f7

be.log
`
PreCatch error code:11, [E11] Memory limit exceeded:[MEM_LIMIT_EXCEEDED]failed alloc size 32.00 KB, exceeded tracker:<EngineAlterTabletTask#baseTabletId=4181081:newTabletId=4188237>, limit 2.00 GB, peak used 2.00 GB, current used 2.00 GB, exec node:<>, execute msg:Allocator mem tracker check failed. backend 172.22.64.33 process memory used 16.25 GB, limit 100.02 GB. Can `set exec_mem_limit=8G` to change limit, details see be.INFO.
0. /root/src/doris-2.0/be/src/common/stack_trace.cpp:302: StackTrace::tryCapture() @ 0x000000000ba1f197 in /opt/app/apache-doris-2.0.2/be/lib/doris_be
`
session
`
SET  exec_mem_limit =32g;
alter table cml.ads_rollup_agg1 add rollup rollup_p_dt(p_dt,rqst_cnt,estd_inc,rqst_id)
`
执行以上语句还是会出现be.log错误

**Reply by yuanyuan8983** [reply]
Please add my wechat, let's look at this problem, my wechat is cyllyy810222

**Comment by BiteTheDDDDt** [comment]
这块的报错信息不太友好，如果查看be日志你应该会看到关于内存不足的报错信息，实际应该在be conf修改memory_limitation_per_thread_for_schema_change_bytes来增大内存限制。
相关内容已经在master做了改进，后续版本会同步相关改动。

**Reply by rotkang** [reply]
谢谢，通过调整be.conf memory_limitation_per_thread_for_schema_change_bytes参数，问题已解决。

**Comment by rotkang** [comment, marked as answer]
问题已解决
